test/functional/ui/statusline_spec.lua

```diff
@@ -799,4 +806,30 @@ describe('default statusline', function()
                                                                   |
     ]])
   end)
+
+  it('shows busy status when buffer is set to be busy', function()
+    exec_lua("vim.o.statusline = ''")
+
+    screen:expect([[
+      ^                                                            |
+      {1:~                                                           }|*13
+      {3:[No Name]                                 0,0-1          All}|
+                                                                  |
+    ]])
+    exec_lua('vim.o.busy = 1')
+    screen:expect([[
+      ^                                                            |
+      {1:~                                                           }|*13
+      {3:[No Name]                               ◐ 0,0-1          All}|
```

Comment:
May want to shift the busy indicator to the LFS, just right of the `[-][RO]` when `set readonly nomodifiable` is set.

For now, we can see how it feels.